docs/docs/08.1-more-about-refs.md

```diff
@@ -1,75 +1,76 @@
-    getInitialState: function() {
-      return {userInput: ''};
-    },
-    handleChange: function(e) {
-      this.setState({userInput: e.target.value});
-    },
-    clearAndFocusInput: function() {
-      this.setState({userInput: ''}); // Clear the input
-      // We wish to focus the <input /> now!
-    },
-    render: function() {
-      return (
-        <div>
-          <div onClick={this.clearAndFocusInput}>
-            Click to Focus and Reset
-          </div>
-          <input
-            value={this.state.userInput}
-            onChange={this.handleChange}
-          />
-        </div>
-      );
-    }
-  });
+## The ref returned from React.render
+
+No to be confused with the `render()` function that you define on your component (and which returns a virtual DOM element), [React.render()](/react/docs/top-level-api.html#react.render) will return a reference to your component's **backing instance** (or null for [stateless components](/react/docs/reusable-components.html#stateless-functions)).
+
+
+```js
+var myComponent = React.render(<MyComponent />, myContainer);
 ```
 
+Keep in mind, however, that the JSX doesn't return a component instance! It's just a **ReactElement**: a lightweight representation that tells React what the mounted component should look like.
 
-Notice how, in this example, we want to "tell" the input something - something that it cannot infer from its props over time. In this case we want to "tell" it that it should now become focused. However, there are some challenges. What is returned from `render()` is not your actual composition of "child" components, but rather it is a *description* of the children at a particular instance in time - a snapshot.
+```js
+var myComponentElement = <MyComponent />; // This is just a ReactElement.
+
+// Some code here...
+
+var myComponentInstance = React.render(myComponentElement, myContainer);
+myComponentInstance.doSomething();
+```
 
 > Note:
 >
-> Remember, what you return from `render()` is not your *actual* rendered children instances. What you return from `render()` is merely a *description* of the children instances in your component's sub-hierarchy at a particular moment in time.
+> This should only ever be used at the top level. Inside components, let your `props` and `state` handle communication with child components, or use one of the other methods of getting a ref (string attribute or callbacks).
+
 
+## The ref Callback Attribute
 
-This means that you should never "hold onto" something that you return from `render()` and then expect it to be anything meaningful.
+React supports a very special attribute that you can attach to any component.  The `ref` attribute can be a callback function, and this callback will be executed immediately after the component is mounted.  The referenced component will be passed in as a parameter, and the callback function may use the component immediately, or save the reference for future use (or both).
```

Comment:
Is it "very" special or just special?

Read: I don't really like the use of "very" here.